src/serializers/type_serializers/json_or_python.rs

```diff
@@ -0,0 +1,76 @@
+use std::borrow::Cow;
+
+use pyo3::intern;
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+
+use super::{BuildSerializer, CombinedSerializer, Extra, TypeSerializer};
+use crate::build_context::BuildContext;
+use crate::build_tools::SchemaDict;
+
+#[derive(Debug, Clone)]
+pub struct JsonOrPythonSerializer {
+    json: Box<CombinedSerializer>,
+    python: Box<CombinedSerializer>,
```

Comment:
I think we should use one of the schemas for all cases here, I guess python but I'm neutral.

The point is that we kind of commit to `json.loads(serializer.to_json(x)) == serializer.to_python(x, mode='json')`, so it doesn't make sense to both here.

Reply:
I think that means we only need to implement `BuildSerializer` and return the serializer for the `python` schema.

Reply:
I'm I'm wrong about this, and we do need both serializers, then we should use the `json` one in the `to_python` method if `mode='json'`.

Reply:
The latter makes sense to me. My thought was that this would be used in two ways:

1) core_schema = json_or_python(…)
2) core_schmea = something(…, serialization=json_or_python(…))

Does that make a difference?

Reply:
Oh makes sense, then do this.